app/controllers/course_summaries_controller.rb

```diff
@@ -10,38 +10,42 @@ def index
   end
 
   def populate
-    visible_assessments_info = {}
-    marking_schemes = {}
+    data = get_table_json_data(current_user)
+    columns = populate_columns
+    average_data = columns.map { |h| [h[:Header], nil] }.to_h
+    median_data = columns.map { |h| [h[:Header], nil] }.to_h
+    individual_data = columns.map { |h| [h[:Header], nil] }.to_h
+    labels = columns.each_with_index.map { |h, i| [h[:Header], i] }.to_h
+
+    visible_assessments = Assessment.order(id: :asc)
+    visible_assessments = visible_assessments.where(is_hidden: false) if current_user.student?
+
+    visible_assessments.each do |a|
+      info = assessment_overview(a)
+      average_data[a.short_identifier] = info[:average]
+      median_data[a.short_identifier] = info[:median]
+      columns[labels[a.short_identifier]][:Header] += " / (#{info[:total].to_i})"
+    end
+
     if current_user.admin?
-      visible_assessments = Assessment.all.order(id: :asc)
-      released_ids = visible_assessments.ids
       MarkingScheme.all.each do |m|
         grades = m.students_weighted_grades_array(current_user)
-        marking_schemes[m.name] = { average: DescriptiveStatistics.mean(grades).round(2),
-                                    median: DescriptiveStatistics.median(grades).round(2) }
+        average_data[m.name] = DescriptiveStatistics.mean(grades).round(2)
```

Comment:
This part is interesting. There's nothing preventing the marking scheme names to be the same as an assessment short identifier (this was avoided in the previous version by keeping the marking scheme stats separate). You can decide how you want to handle this!